x/merkledb/db.go

```diff
@@ -51,11 +51,11 @@ var (
 	intermediateNodePrefix = []byte{2}
 
 	cleanShutdownKey        = []byte(string(metadataPrefix) + "cleanShutdown")
+	rootDBKey               = []byte(string(metadataPrefix) + "root")
```

Comment:
realized belatedly that you don't actually need this since the root should always be the smallest key present in your nodes db.  You should be able to just grab the first key on the iterator, right?

Reply:
We could do that, but I feel like this is a bit cleaner than having to check both the value and intermediate node databases

Reply:
I'm pretty sure what you have now is broken then, since you only record the key and not which db it is in.